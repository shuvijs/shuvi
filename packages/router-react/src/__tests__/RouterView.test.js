import { create as createTestRenderer } from 'react-test-renderer';
import { MemoryRouter as Router, RouterView } from '..';

describe('useOutlet', () => {
  describe('when there is no child route', () => {
    it('returns null', () => {
      function Home() {
        return <RouterView />;
      }

      let renderer = createTestRenderer(
        <Router
          initialEntries={['/home']}
          routes={[
            {
              path: '/home',
              component: Home
            }
          ]}
        >
          <RouterView />
        </Router>
      );

      expect(renderer.toJSON()).toBeNull();
    });
  });

  describe('when there is a child route', () => {
    it('returns an element', () => {
      function Users() {
        return <RouterView />;
      }

      function Profile() {
        return <p>Profile</p>;
      }

      let renderer = createTestRenderer(
        <Router
          initialEntries={['/users/profile']}
          routes={[
            {
              path: '/users',
              component: Users,
              children: [
                {
                  path: '/profile',
                  component: Profile
                }
              ]
            }
          ]}
        >
          <RouterView />
        </Router>
      );

      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
});
