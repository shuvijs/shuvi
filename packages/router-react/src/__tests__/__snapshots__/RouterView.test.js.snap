// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOutlet when there is a child route returns an element 1`] = `
<p>
  Profile
</p>
`;
